### PR TITLE
[Fix][Worker] alert email in sql task in K8S

### LINKED ISSUE #12088: [Bug] [Worker] alert email in sql task in K8S
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

dolphinscheduelr deploy in k8s.

1. Create a sql task in workflow definition
     - Datasource types: MYSQL
     - SQL_TYPE: Query
     - Send Email: Enable
2. After start the workflow definition,  no email receive, no ERROR in task log.

### What you expected to happen

receive the Email

### How to reproduce

1. Create a sql task in workflow definition
     - Datasource types: MYSQL
     - SQL_TYPE: Query
     - Send Email: Enable
2. After start the workflow definition,  no email receive, no ERROR in task log.

find ERROR in log of the worker server
```
[WARN] 2022-09-21 13:15:48.196 +0800 org.apache.dolphinscheduler.remote.NettyRemotingClient:[368] - [WorkflowInstance-3179][TaskInstance-30336] - connect to Host{address='localhost:50052', ip='localhost', port=50052} error
io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: localhost/127.0.0.1:50052
Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
 at io.netty.channel.unix.Errors.throwConnectException(Errors.java:124)
 at io.netty.channel.unix.Socket.finishConnect(Socket.java:251)
 at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:673)
 at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:650)
 at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:530)
 at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:465)
 at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378)
 at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
 at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
 at java.lang.Thread.run(Thread.java:750)
[ERROR] 2022-09-21 13:15:48.196 +0800 org.apache.dolphinscheduler.service.alert.AlertClientService:[116] - [WorkflowInstance-3179][TaskInstance-30336] - sync alert send error
org.apache.dolphinscheduler.remote.exceptions.RemotingException: connect to : Host{address='localhost:50052', ip='localhost', port=50052} fail
 at org.apache.dolphinscheduler.remote.NettyRemotingClient.sendSync(NettyRemotingClient.java:257)
 at org.apache.dolphinscheduler.service.alert.AlertClientService.sendAlert(AlertClientService.java:111)
 at org.apache.dolphinscheduler.service.alert.AlertClientService.sendAlert(AlertClientService.java:93)
 at org.apache.dolphinscheduler.server.worker.runner.TaskExecuteThread.sendAlert(TaskExecuteThread.java:240)
 at org.apache
```

### Anything else

_No response_

### Version

3.0.0

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by BruceMaa**
worker.alert-listen-host has not get ALERT_LISTEN_HOST environment variable in worker containers env

## Purpose of the pull request

This pull request fix #12088 

## Brief change log

This pull request update deploy/kubernetes/dolphinscheduler/templates/statefulset-dolphinscheduler-worker.yaml

```yaml
      containers:
        - name: {{ include "dolphinscheduler.fullname" . }}-worker
          env:
            - name: WORKER_ALERT_LISTEN_HOST
```
## Verify this pull request

This pull request is code cleanup without any test coverage.

### CODE REVIEW

File: dolphinscheduler-worker/src/main/resources/application.yaml [hunk 1]
```diff
@@ -76,7 +76,7 @@ worker:
   groups:
     - default
   # alert server listen host
-  alert-listen-host: localhost
+  alert-listen-host: ${ALERT_LISTEN_HOST:localhost}
```
Comment (caishunfeng) [inline dolphinscheduler-worker/src/main/resources/application.yaml]: I think we can use the env variable to replace it, like: https://github.com/apache/dolphinscheduler/blob/3.0.0-release/script/env/dolphinscheduler_env.sh#L31    WDYT? 
cc @kezhenxu94

Comment (BruceMaa) [reply dolphinscheduler-worker/src/main/resources/application.yaml]: how about change https://github.com/apache/dolphinscheduler/blob/9badb2d2fbc6b79efca96247dbc49a39ca6868f7/deploy/kubernetes/dolphinscheduler/templates/statefulset-dolphinscheduler-worker.yaml#L67

to

```
  env:
    - name: WORKER_ALERT_LISTEN_HOST
```

@caishunfeng

Comment (caishunfeng) [reply dolphinscheduler-worker/src/main/resources/application.yaml]: > how about change
> 
> https://github.com/apache/dolphinscheduler/blob/9badb2d2fbc6b79efca96247dbc49a39ca6868f7/deploy/kubernetes/dolphinscheduler/templates/statefulset-dolphinscheduler-worker.yaml#L67
> 
> to
> 
> ```
>   env:
>     - name: WORKER_ALERT_LISTEN_HOST
> ```
> 
> @caishunfeng

LGTM if it works.

**Comment by caishunfeng** [comment]
rerun the fail CI

**Comment by simsicon** [comment]
I can confirm this pr can fix the issue. can reviewers try to review it again?

**Review by caishunfeng — APPROVED** [review]
+1

**Review by fuchanghai — APPROVED** [review]

**Comment by caishunfeng** [comment]
Seems the CI `Mergeable: milestone-label-check` block, but how to check the detail or rerun it? @zhongjiajie

**Comment by zhongjiajie** [comment]
@BruceMaa can you rebase on dev and then push to restart the CI?

**Comment by BruceMaa** [comment]
> 

sorry, I did it wrong, now open another PR. #12369 

@zhongjiajie @caishunfeng @fuchanghai

**Pull request closed without merging**